multidict/_pair_list.c

```diff
@@ -612,40 +669,48 @@ pair_list_pop_one(PyObject *op, PyObject *ident, PyObject *key)
             if (pair_list_del_at(list, pos) < 0) {
                 goto fail;
             }
+            Py_DECREF(ident);
             return value;
         }
         else if (tmp < 0) {
-            return NULL;
+            goto fail;
         }
     }
 
     PyErr_SetObject(PyExc_KeyError, key);
-    return NULL;
+    goto fail;
```

Comment:
This "goto" not needed

Reply:
I think this style is a little cleaner (more readable) than `goto` omitting.
Generated code should be the same.